### PR TITLE
fix typos in documentation

### PR DESCRIPTION
**Pull request by striezel**
This is to fix several typos, mostly in documentation files.

The pull request does not change any functionality or code coverage, because it does not change code in the first place.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ All notable changes to this project will be documented in this file. This projec
 
 [Full Changelog](https://github.com/nlohmann/json/compare/v3.10.2...v3.10.3)
 
-- Parsing an emtpy string returns a string with size 1 instead of expected 0 [\#3057](https://github.com/nlohmann/json/issues/3057)
+- Parsing an empty string returns a string with size 1 instead of expected 0 [\#3057](https://github.com/nlohmann/json/issues/3057)
```
Comment (nlohmann) [inline ChangeLog.md:34]: This file is generated from the titles of Github issues with every release. The typos here are already in Github.

Please revert the changes to this file.

**Review by nlohmann — CHANGES_REQUESTED** [review]

File: ChangeLog.md (hunk 1 above)
Comment (striezel) [reply ChangeLog.md]: Done. Commit 1ea1040ec reverts the changes made to the ChangeLog.md file.

**Review by nlohmann — APPROVED** [review]
Looks good to me.

**Comment by nlohmann** [comment]
Thanks a lot!
